### PR TITLE
remove error parameter from trace, debug & info

### PR DESCRIPTION
**Pull request by weissi**
CC @tomerd

### CODE REVIEW

**Review by ktoso — APPROVED** [review]
LGTM 👍 Good catch

**Comment by ktoso** [comment]
(also: very good to allow attaching error in `warning` level, was a mistake we had in some apis in the past)

File: Sources/Logging/Logging.swift [hunk 1]
```diff
@@ -34,25 +34,25 @@ public struct Logger {
     }
 
     @inlinable
-    func log(level: Logging.Level, message: @autoclosure () -> String, metadata: @autoclosure () -> Logging.Metadata? = nil, error: Error? = nil, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+    public func log(level: Logging.Level, message: @autoclosure () -> String, metadata: @autoclosure () -> Logging.Metadata? = nil, error: Error? = nil, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
```
Comment (tomerd) [inline Sources/Logging/Logging.swift:37]: :+1:

**Review by tomerd — COMMENTED** [review]
@ktoso @weissi my first instinct was also to only support it for warn and error, so i doubled checked what slf4j does and they support it in all variants. see https://www.slf4j.org/api/org/slf4j/Logger.html

no strong opinion, but worth pointing out why i did it this way. wdyt?

File: Sources/Logging/Logging.swift (hunk 1 above)
Comment (weissi) [reply Sources/Logging/Logging.swift:37]: oops, that slipped in but whatever, let's do it in this PR :)

**Comment by weissi** [comment]
@tomerd I'd be in favour of not having it everywhere because if people _really_ want to, they can create that through an extension, using the `logger.log` method which does take an error.

But I'm not super strong here. @ktoso ?

**Comment by ktoso** [comment]
No strong opinion, as long as warning has it that’s good. (That’s a mistake Akka made where it’s logging api did not allow exceptions to be passed in warning level which really annoyed a lot of people)

I suppose there could be uses to log an error even on trace for things like “this failed but we’re trying again” only to after some errors log an actual errror...

**Comment by ktoso** [comment]
But people could add these with an extension if it is needed hm

**Comment by dhoepfl** [comment]
Could you explain why you want to remove it? After all, you can just leave it out on call side.

I, too, think that it is not a uncommon use case to include errors in your log that do not pose a problem:
```
storeEventLocally(…)
do {
   try sendPendingEventsToServer()
} catch {
   logger.debug("Failed to sync to server, will be sent as part of next sync", error: error)
   // Could also be trace or info.
}
```

**Comment by ktoso** [comment]
I think I'm happy to keep the param around, good point @dhoepfl.
Let's see what Johannes thinks.

My main concern was to have ti available on warning at least; I don't mind having it on others.

**Comment by weissi** [comment]
@dhoepfl / @ktoso yeah, let's keep it around, doesn't hurt.

**Pull request closed without merging**